runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -603,13 +606,40 @@ InterpreterEmulator::visitInvokedynamic()
    TR::Node *callNode = 0;
    TR::ResolvedMethodSymbol *resolvedSymbol = 0;
    Operand *result = NULL;
-
-   TR_ResolvedMethod * owningMethod = _methodSymbol->getResolvedMethod();
+   int32_t callSiteIndex = next2Bytes();
    TR::KnownObjectTable *knot = comp()->getOrCreateKnownObjectTable();
-   if (knot && !owningMethod->isUnresolvedCallSiteTableEntry(cpIndex))
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   if (!knot && owningMethod->isUnresolvedCallSiteTableEntry(callSiteIndex)) return;
+
+   // add appendix object to knot and push to stack
+   uintptr_t appendixAddress = owningMethod->appendixAddressFromInvokeDynamicSideTable(callSiteIndex);
+   push(new (trStackMemory()) KnownObjOperand(knot->getOrCreateIndexAt(&appendixAddress)));
```

Comment:
If `appendixAddressFromInvokeDynamicSideTable` returns the address instead of the appendix object reference, we don't need to take the address of it.

If `knot` is null, just push unknown operand.